### DISCUSSION TITLE
Errors regarding checkpoints_v2 and openvoice_app.py

### DISCUSSION THREAD

**Discussion by syfai** [myshell-ai/OpenVoice#321] [General]
Hello People please help me how do I run this model successfully.
I'm totally non technical and very frustrated with all those errors, so please somebody help me out. Following is the refined code given by chatgpt after working several hours and chats that i researched for as an non technical coding layman

Here is the code I'm trying to run through VPS with Debian 12  as OS (CPU)
First, logged into my VPS via SSH in Terminal
sudo apt update
sudo apt upgrade -y
sudo apt install -y python3 python3-pip git
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
bash Miniconda3-latest-Linux-x86_64.sh
source ~/.bashrc
conda create -n openvoice python=3.9
conda activate openvoice
git clone https://github.com/myshell-ai/OpenVoice.git
cd OpenVoice
pip install -e .
exit
(Downloaded the checkpoints_v2)
scp -r C:\Users\name\Downloads\checkpoints_v2 root@ip_address:/root/OpenVoice/ 
ssh root@ip_address 
ls /root/OpenVoice
ls /root/OpenVoice/checkpoints_v2
conda activate openvoice
python -m openvoice.openvoice_app --share

now after doing all these there are errors such as follows,
FileNotFoundError: [Errno 2] No such file or directory: 'checkpoints/base_speakers/EN/config.json'
FileNotFoundError: [Errno 2] No such file or directory: '/root/OpenVoice/checkpoints/base_speakers/ZH/config.json'
etc.,.

and the structure of the downloaded checkpoints_v2 is like this,
checkpoints_v2/
├── base_speakers
│   ├── ses
│   │   ├── en-au.pth
│   │   ├── en-br.pth
│   │   ├── en-default.pth
│   │   ├── en-india.pth
│   │   ├── en-newest.pth
│   │   ├── en-us.pth
│   │   ├── es.pth
│   │   ├── jp.pth
│   │   ├── kr.pth
│   │   ├── zh.pth
│   │   ├── fr.pth
├── converter
│   ├── config.json
│   └── checkpoint.pth

so what should be the correct structure of this folder checkpoints_v2, to run successfully this model. and chatgpt was mentioning some execution errors in openvoice/openvoice_app.py, so are there any errors in this file or any other file apart from this checkpoints_v2 structer. Please Help Soon.

**Comment by loscrossos** [comment]
hi, i just got it working after a day of installation battle :D. 

there are several things that i see problematic:

couple of thoughts:
you say you are on debian linux.. but your example says something about "C:users\.." which is a windows path.. are you on a virtual machine? VMs dont support CUDA. just so you know.

the first real problem i see is that you have to change the code to point to where you downloaded the model.

your error is looking for 

```
/root/OpenVoice/**checkpoints**/base_speakers/ZH/config.json
```

but your model is under checkpoints**_v2**. so that can not work in the first place.

you need a full path of the real model to be able to start at all.

you can check this easily:

on a terminal run "ls [your path]" for example
```
ls /root/OpenVoice/checkpoints/base_speakers
```
 and you should see "base speakers" and "converter". but i think that path will not work as you get errors with it.

lets assume you have the right path (im making this path up.. you need to find the right one!) 
/root/OpenVoice/checkpoints_v2/base_speakers


i think the notebooks have easier code to try for the first time. you are using the gradio app, which is nice to use but has more dependencies and is a bit harder to setup.

anyway, lets continue with the app.. but the notebook example has the same requirement.
you have now to at least changet the model variables in the code. 
in your example of the file openvoice.openvoice_app:

change the lines 14 and 15:
```
en_ckpt_base = 'checkpoints/base_speakers/EN'
zh_ckpt_base = 'checkpoints/base_speakers/ZH'
```
to have the full path of your model directory:
```
en_ckpt_base = '/root/OpenVoice/checkpoints_v2/base_speakers/EN'
zh_ckpt_base = '/root/OpenVoice/checkpoints_v2/base_speakers/ZH'
```
hope this helps!

**Reply by syfai** [reply]
Oh thanks a lot @javaGH. Got it!
